drivers/random/random_stm32_rng.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2017 Erwin Rol <erwin@erwinrol.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <device.h>
+#include <random.h>
+#include <drivers/rand32.h>
+#include <init.h>
+#include <misc/__assert.h>
+#include <misc/util.h>
+#include <errno.h>
+#include <soc.h>
+#include <misc/printk.h>
+#include <clock_control.h>
+#include <clock_control/stm32_clock_control.h>
+
+#include <stm32f4xx_hal_rng.h>
+
+#include "random_stm32_rng_priv.h"
+
+static int random_stm32_rng_get_entropy(struct device *dev, u8_t *buffer,
+					u16_t length)
+{
+	struct random_stm32_rng_dev_data *dev_data;
+	u32_t rndbits;
+
+	__ASSERT_NO_MSG(dev != NULL);
+	__ASSERT_NO_MSG(buffer != NULL);
+
+	dev_data = DEV_DATA(dev);
+
+	__ASSERT_NO_MSG(dev_data != NULL);
+
+	while (length > 0) {
+
+		if (HAL_RNG_GenerateRandomNumber(&(dev_data->hrng), &rndbits) != HAL_OK) {
+			return -EIO;
+		}
+
+		switch (length) {
```

Comment:
This seems like we can be a lot smarter about how we take rndbits and put it into the buffer, like if length > 4, do 4 bytes at a time.